src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -506,6 +507,89 @@ protected function initializeBundles()
     }
 
     /**
+     * Returns an array of bundles to register, with dependencies, ordered
+     *
+     * @return \Symfony\Component\HttpKernel\Bundle\BundleInterface[] An array of bundle instances.
+     */
+    protected function orderedRegisteredBundlesAndDependencies()
+    {
+        $bundles = $this->registerBundles();
+        $topMostBundles = $topMostBundlesMap = array();
+        $hasDependencies = false;
+
+        // Build up bundles as a hash with FQN for basis to rebuild again in correct order given dependencies
+        foreach ($bundles as $bundle) {
+            $topMostBundles[] = $bundleFQN = get_class($bundle);
+            $topMostBundlesMap[$bundleFQN] = $bundle;
+
+            if ($bundle instanceof BundleDependenciesInterface) {
+                $hasDependencies = true;
+            }
+        }
+
+        if (!$hasDependencies) {
+            return $bundles;
+        }
+
+        // Rebuild bundles order with dependencies recursively
+        $bundles = array();
+        $this->appendDependenciesRecursively(
+            $topMostBundles,
+            $bundles,
+            $topMostBundlesMap,
+            get_class($this)
+       );
+
+        return $bundles;
+    }
+
+    /**
+     * Append dependencies of bundles recursively
+     *
+     * Accepted arguments are as documented below where `string` implies a string with class FQN(Fully Qualified Name),
+     * this string must be in exactly same format as returned by get_class() and PHP 5.5's CLASS constant.
+     * Example of FQN string: "Symfony\Component\HttpKernel\Bundle\BundleInterface"
+     *
+     * @param string[] $directChildren Dependencies to apply
+     * @param \Symfony\Component\HttpKernel\Bundle\BundleInterface[<string>] $bundles Ordered bundles to append dependencies to
+     * @param \Symfony\Component\HttpKernel\Bundle\BundleInterface[<string>] $topMostBundlesMap Loaded Bundles from registerRootBundles()
+     * @param string $parent For use in exception message if a dependency can not be loaded.
+     *
+     * @throws \Exception
+     */
+    protected function appendDependenciesRecursively(array $directChildren, array &$bundles, array $topMostBundlesMap, $parent)
+    {
+        while (!empty($directChildren)) {
+            $dependencyFQN = array_shift($directChildren);
+            if (isset($bundles[$dependencyFQN])) {
+                continue;
+            }
+
+            // If already loaded root bundle, use that to not re instantiate
+            if (isset($topMostBundlesMap[$dependencyFQN])) {
+                $dependency = $topMostBundlesMap[$dependencyFQN];
+            } else {
+                if (!class_exists($dependencyFQN)) {
+                    throw new \RuntimeException(sprintf("Could not find Bundle '%s', set as dependency by '%s'", $dependencyFQN, $parent));
```

Comment:
Found `ClassNotFoundException`, seems to be the most fitting here.
@KingCrunch What do you think?